core/domain/summary_services.py

```diff
@@ -163,6 +163,52 @@ def get_displayable_collection_summary_dicts_matching_ids(collection_ids):
     return _get_displayable_collection_summary_dicts(collection_summaries)
 
 
+def get_collection_node_exploration_dicts(
+        exploration_ids, editor_user_id=None):
+    """Given a list of exploration ids, optionally filters the list for
```

Comment:
Use more extensive docstrings. It's important here to highlight what exactly is being returned, too, since this "metadata dict" is a new concept.